src/misc/cross.cpp

```diff
@@ -16,26 +16,108 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
  */
 
-
-#include "dosbox.h"
 #include "cross.h"
-#include "support.h"
+
 #include <string>
+#include <vector>
+
 #include <limits.h>
 #include <stdlib.h>
+#include <sys/types.h>
+#include <unistd.h>
 
 #ifdef WIN32
 #ifndef _WIN32_IE
 #define _WIN32_IE 0x0400
 #endif
 #include <shlobj.h>
+#else
+#include <libgen.h>
 #endif
 
-#if defined HAVE_SYS_TYPES_H && defined HAVE_PWD_H
-#include <sys/types.h>
+#if defined HAVE_PWD_H
 #include <pwd.h>
 #endif
 
+#include "support.h"
+
+#ifndef WIN32
+
+std::string cached_conf_path;
+
+static bool CreateDirectories(const std::string &path)
+{
+	struct stat sb;
+	if (stat(path.c_str(), &sb) == 0) {
+		const bool is_dir = ((sb.st_mode & S_IFMT) == S_IFDIR);
+		return is_dir;
+	}
+
+	std::vector<char> tmp(path.length());
+	std::copy(path.begin(), path.end(), tmp.begin());
+	std::string dname = dirname(tmp.data());
+
+	// Create parent directories recursively
+	if (!CreateDirectories(dname))
+		return false;
+
+	return (mkdir(path.c_str(), 0700) == 0);
+}
+
+static std::string GetConfigName()
+{
+	std::string file_name;
+	Cross::GetPlatformConfigName(file_name);
+	return file_name;
```

Comment:
If `GetPlatformConfigName` is self-sufficient and doesn't need any inputs to determine config name, then suggest making take in nothing, `GetPlatformConfigName()` and returning the as-determined string.